airflow/models.py

```diff
@@ -336,7 +336,8 @@ def kill_zombies(self, session):
                 if ti.task_id in dag.task_ids:
                     task = dag.get_task(ti.task_id)
                     ti.task = task
-                    ti.handle_failure("{} killed as zombie".format(ti))
+                    ti.test_mode = False
```

Comment:
OK, found the issue, actually `TaskInstance` `init` is not being called.. as explained here http://stackoverflow.com/questions/17256604/python-class-init-not-working